release/serve_tests/workloads/microbenchmarks.py

```diff
@@ -158,6 +167,69 @@ async def _main(
             )
             serve.shutdown()
 
+        if run_streaming:
+            # Direct streaming between replica
+            serve.run(
+                Streamer.options(max_ongoing_requests=1000).bind(
+                    tokens_per_request=STREAMING_TOKENS_PER_REQUEST,
+                    inter_token_delay_ms=10,
+                )
+            )
+            mean, std, latencies = await run_throughput_benchmark(
+                fn=partial(
+                    do_single_http_batch,
+                    batch_size=STREAMING_HTTP_BATCH_SIZE,
+                    stream=True,
+                    record_latencies=True,
+                ),
+                multiplier=STREAMING_HTTP_BATCH_SIZE * STREAMING_TOKENS_PER_REQUEST,
+                num_trials=STREAMING_NUM_TRIALS,
+                # Complete only one batch of requests
```

Comment:
Hmm, I'm not too sure why is this one batch of requests. Is it bc `do_single_http_batch()` is used?

Reply:
It's because `run_throughput_benchmark` repeatedly calls `do_single_http_batch()` anyways, and each call takes 10+s (we are adding delays on replica side), so it's more straightforward to just do a single batch per trial and calculate average throughput across trials that way.

Reply:
Yes, I understand and agreed with the approach. But the placement of this comment can probably be moved up to line 183 to make it clear what determines the batch (or even above line 181 and add more explanation like you stated here). Being here feel like `trial_runtime=10` determine batch. Sorry not being more clear on the commend 🙏

Reply:
ah got it, makes sense I've added this explanation as a comment